Sources/RealDeviceMap/Misc/PVPStatsManager.swift

```diff
@@ -0,0 +1,324 @@
+            if id.starts(with: "V"), id.contains(string: "_POKEMON_"),
+                let pokemonInfo = template["pokemon"] as? [String: Any],
+                let pokemonName = pokemonInfo["uniqueId"] as? String,
+                let statsInfo = pokemonInfo["stats"] as? [String: Any],
+                let baseStamina = statsInfo["baseStamina"] as? Int,
+                let baseAttack = statsInfo["baseAttack"] as? Int,
+                let baseDefense = statsInfo["baseDefense"] as? Int {
+                guard let pokemon = pokemonFrom(name: pokemonName) else {
+                    Log.warning(message: "[PVPStatsManager] Failed to get pokemon for: \(pokemonName)")
+                    return
+                }
+                let formName = pokemonInfo["form"] as? String
+                let form: POGOProtos_Enums_Form?
+                if let formName = formName {
+                    guard let formT = formFrom(name: formName) else {
+                        Log.warning(message: "[PVPStatsManager] Failed to get form for: \(formName)")
+                        return
+                    }
+                    form = formT
+                } else {
+                    form = nil
+                }
+                var evolutions = [PokemonWithForm]()
+                let evolutionsInfo = pokemonInfo["evolutionBranch"] as? [[String: Any]] ?? []
+                for info in evolutionsInfo {
+                    if let pokemonName = info["evolution"] as? String, let pokemon = pokemonFrom(name: pokemonName) {
+                        let formName = info["form"] as? String
+                        let form = formName == nil ? nil : formFrom(name: formName!)
+                        evolutions.append(.init(pokemon: pokemon, form: form))
+                    }
+                }
+                let stats = Stats(baseAttack: baseAttack, baseDefense: baseDefense,
+                                  baseStamina: baseStamina, evolutions: evolutions)
+                self.stats[.init(pokemon: pokemon, form: form)] = stats
+            }
+        }
+    }
+
+    internal func getPVPStats(pokemon: POGOProtos_Enums_PokemonId, form: POGOProtos_Enums_Form?, iv: IV, level: Double,
+                              league: League) -> (ranking: Int, percentage: Double)? {
+        guard let stats = getTopPVP(pokemon: pokemon, form: form, league: league, level: level) else {
+            return nil
+        }
+        guard let index = stats.firstIndex(where: { value in
+            for ivlevel in value.ivs where ivlevel.0 == iv {
+                return true
+            }
+            return false
+        }) else {
+            return nil
+        }
+        let max = Double(stats[0].value)
+        let value = Double(stats[index].value)
+        return (index + 1, value/max)
+    }
+
+    internal func getPVPStatsWithEvolutions(pokemon: POGOProtos_Enums_PokemonId, form: POGOProtos_Enums_Form?,
```

Comment:
It would be nice if this excluded evolutions that are impossible due to the pokemon's costume.
Costumes that include 'Noevolve' in the string, like 10 -> 17 and 22 -> 25, can't evolve.
Current Example:
Pikachu with Costume #25, sinnoh2020Noevolve, should not consider Raichu as it can not evolve.

Reply:
Done. Now ignores no-evolve forms